src/components/countryDetails/CountryDetails.tsx

```diff
@@ -1,45 +1,214 @@
+						});
+
+						const currencieKey = Object.keys(currencies)[0];
+
+						const currencieName =
+							currencies[currencieKey as unknown as number].name;
+
+						const languagesKeys = Object.keys(languages);
+
+						return (
+							<div key={name.common}>
+								<StyledDivWithCountryInfo>
+									<StyledLeftColumnWithCountryInfo>
+										<StyledCountryName>{name.common}</StyledCountryName>
+										<StyledParagraphWithCountryInfo>
+											<StyledSpanWithCategoryName>
+												Native Name:
+											</StyledSpanWithCategoryName>{" "}
+											{altSpellings[2]}
+										</StyledParagraphWithCountryInfo>
+
+										<StyledParagraphWithCountryInfo>
+											<StyledSpanWithCategoryName>
+												Population:
+											</StyledSpanWithCategoryName>{" "}
+											{population}
+										</StyledParagraphWithCountryInfo>
+										<StyledParagraphWithCountryInfo>
+											<StyledSpanWithCategoryName>
+												Region:
+											</StyledSpanWithCategoryName>{" "}
+											{region}
+										</StyledParagraphWithCountryInfo>
+										<StyledParagraphWithCountryInfo>
+											<StyledSpanWithCategoryName>
+												Sub Region:
+											</StyledSpanWithCategoryName>{" "}
+											{subregion}
+										</StyledParagraphWithCountryInfo>
+										<StyledParagraphWithCountryInfo>
+											<StyledSpanWithCategoryName>
+												Capital:
+											</StyledSpanWithCategoryName>{" "}
+											{capital}
+										</StyledParagraphWithCountryInfo>
+									</StyledLeftColumnWithCountryInfo>
+									<div>
+										<StyledParagraphWithCountryInfo>
+											<StyledSpanWithCategoryName>
+												Top Level Domain:
+											</StyledSpanWithCategoryName>{" "}
+											{tld[0]}
+										</StyledParagraphWithCountryInfo>
+										<StyledParagraphWithCountryInfo>
+											<StyledSpanWithCategoryName>
+												Currencies:
+											</StyledSpanWithCategoryName>{" "}
+											{currencieName}
+										</StyledParagraphWithCountryInfo>
+										<StyledParagraphWithCountryInfo>
+											<StyledSpanWithCategoryName>
+												Languages:
+											</StyledSpanWithCategoryName>{" "}
+											{languagesKeys.map((lang) => {
+												return (
+													<span key={lang}>
+														{languages[lang as unknown as number]},{" "}
+													</span>
+												);
+											})}
+										</StyledParagraphWithCountryInfo>
+									</div>
+								</StyledDivWithCountryInfo>
+								<StyledDivWithButtons>
+									<StyledSpanWithCategoryName>
+										Border Countries:
+									</StyledSpanWithCategoryName>{" "}
+									{arr.length
+										? arr.map((country, index) => {
+												return (
+													<StyledLinkWithBorderCountries
+														to={`/${country?.name.common}`}
+														key={index}
+														type="button"
+													>
+														{country?.name.common.substr(0, 12)}
```

Comment:
why was this substr used?

Reply:
I removed it. But I used it as some names of countries are too long so buttons do not look good.